index.js

```diff
@@ -306,4 +306,22 @@ Notifications.clearAllNotifications = function() {
 	return this.callNative('clearAllNotifications', arguments)
 }
 
+Notifications.removeAllDeliveredNotifications = function() {
+  if ( Platform.OS === 'ios' ) {
+    return this.callNative('removeAllDeliveredNotifications')
+  }
+}
+
+Notifications.removeDeliveredNotifications = function() {
+  if ( Platform.OS === 'ios' ) {
+    return this.callNative('removeDeliveredNotifications', arguments)
+  }
+}
+
+Notifications.getDeliveredNotifications = function() {
+  if ( Platform.OS === 'ios' ) {
+    return this.callNative('getDeliveredNotifications', arguments)
+  }
+}
+
 module.exports = Notifications;
```

Comment:
@varungupta85 first you need the delivered notifications, then you need filter to get the notification or notifications you want to delete... Something like this:

```javascript
NotificationsIOS.getDeliveredNotifications((notifications) => {
    let toDelete = notifications.filter((n) => return YOUR FILTER CRITERIA);
    
    NotificationsIOS.removeDeliveredNotifications(toDelete)
});